community/submit_guide/pr-and-code-quality.md

```diff
@@ -0,0 +1,420 @@
+
+- Backend code specification Maven plugin: checkstyle
+Just run mvn `checkstyle:checkstyle` after installing the plugin.
+
+- Frontend code formatting plugin eslint
+
+  The original command is `npx eslint --cache --max-warnings 0 "{src,mock}/**/*.{vue,ts,tsx}" --fix`
+
+  Encapsulated as `npm run lint:eslint`
+
+## 3 Programming Specification
+
+### 3.1 Naming Style
+
+1. Prioritize selecting nouns for variable naming
+   It's easier to distinguish between variables or methods
+   - Example of variables:
+     - `Cache<String> publicKeyCache;`
+
+2. Pinyin abbreviations are prohibited for variables (excluding nouns such as place names)Such as chengdu.
+
+3. It is recommended to end variable names with a type
+   For variables of type Collection/List, take xxxx (plural representing multiple elements) or end with xxxList (specific type)
+   For variables of type map, describe the key and value clearly:
+   Example:
+
+   - `Map<Long, User> idUserMap`
+   - `Map<Long, String> userIdNameMap`
+
+4. That can intuitively know the type and meaning of the variable through its name
+   Method names should start with a verb first as follows:
+
+   `void computeVcores(Object parameter1);`
+
+5. The methods name of basic CRUD of the database layer (non-service layer) should be uniformly standardized according to name com.baomidou.mybatisplus.core.mapper.BaseMapper:
+
+   If perform a database select operation, the name of the method should be started with select
+
+   For example, `selectById`, `selectByXxx`, `selectPageByXxx`
+
+   - If perform a database <mark> update </mark> statement operation, the name of the method should be started with `update`
+   - If perform a database <mark> insert </mark> statement operation, the name of the method should be started with `insert`
+   - If perform a database <mark> delete </mark> statement operation, the name of the method should be started with `delete`
+
+6. The methods name of basic CRUD of the service layer should be named as com.baomidou.mybatisplus.extension.service.IService:
+
+   - If perform a database <mark> select </mark> operation to query multiple records, the name of the method should be started with a `list`, such as `listByIds`, `listByXxx`
+   - If perform a database <mark> select </mark> operation to query a single record, the name of the method should be started with get, such as `getByName` and `getOne`
+   - If perform a database <mark> update </mark> operation, the name of the method should be started with `update`
+   - If perform a database <mark> insert </mark> operation, the name of the method should be started with `save`
+   - If perform a database <mark> delete </mark> operation, the name of the method should be started with `remove`
+
+### 3.2 Constant Definition
+
+Set the serialVersionUID of all classes to 1L, following Flink's serialVersionUID
+Redundant strings should be modified to constants, here are a positive demo and a negative demo:
+
+```java
```

Comment:
I'm very sorry for not being able to maintain a consistent sample style during the draft stage.

Could we maintain a consistent format for both positive and negative examples ?


- don't:
......
- do:
......

`OR`
- negative demo:
.....
- positive demo:
....